### PR TITLE
Added the codeHighlighting using Prismjs

### PR DESCRIPTION
**Pull request by Aditya-eddy**
Fixes the issue [#1527](https://github.com/keploy/keploy/issues/1527)
Description:-
1. Created a prism loader
2. imported the prism loader in the slug.
3. Languages imported (typescript,python,yaml,javascript,golang)

### CODE REVIEW

**Comment by Aditya-eddy** [comment]
Hey @nehagup I'll attach the screenshots of the highlighted codes. Is this fixed according to the issue. Will be working on the copy button. Thanks!!
Screenshot from 2024-02-08 16-09-15
Screenshot from 2024-02-08 16-09-04
Screenshot from 2024-02-08 16-08-25

File: pages/technology/[slug].tsx [hunk 1]
```diff
@@ -13,63 +13,55 @@ import PostTitle from "../../components/post-title";
...
-
+  console.log(post.content);
   if (!router.isFallback && !post?.slug) {
     return <ErrorPage statusCode={404} />;
   }
 
   return (
     <Layout preview={preview}>
+      <Head>
+        <title>{`${post.title} | Next.js Blog Example with ${CMS_NAME}`}</title>
```
Comment (nehagup) [inline pages/technology/[slug].tsx:29]: Title needs to change - ${post.title} | Keploy Blog

**Review by nehagup — COMMENTED** [review]
Thanks for taking this initiative! A couple of things - 

1. For a few code blocks highlighting still doesn't work, example /blog/technology/mongodb-in-mock-mode-acting-the-server-part, blog/technology/go-mocks-and-stubs-made-easy
2. Remove console log to avoid cluttered logs

**Comment by Aditya-eddy** [comment]
Hey @nehagup about highlighting about few blocks as you mentioned above. We can apply prism-loader.tsx if code className specifies on which language the code is written. Here is the html,css for that few block
`<code>func handleConnection(clientConn net.Conn, serverConn net.Conn) {
    // Read requests from the driver client connection
    requestbuffer := make([]byte, 1024)    
    n, _ := clientConn.Read(requestbuffer)

    // write the request to the MongoDB server connection
    serverConn.Write(requestBuffer)

    // Read response from the MongoDB server connection
    responsebuffer := make([]byte, 1024)    
    n, _ := serverConn.Read(responsebuffer)

    // write the response to the driver client connection
    clientConn.Write(responsebuffer)
}</code>`
As we can see above the code doesnt have className with language specified..Hence it's not possible to use code higlighting on them. But if you have any solution I would be waiting on the feedback. Thanks!!

**Review by nehagup — COMMENTED** [review]
This looks good @Aditya-eddy ! Can we keep default language for "no-language" mentioned cases as bash?

**Comment by Aditya-eddy** [comment]
Hey @nehagup I have kept the default language as bash as you'll see the snippet which earlier was with only code tags has been modified with having className bash. Will be waiting on the feeback. Thanks!!
Screenshot from 2024-02-13 21-13-05

**Review by nehagup — COMMENTED** [review]
@Aditya-eddy looks good! Just need to resolve the build err
```


 ✓ Creating an optimized production build
 ✓ Compiled successfully
 ✓ Collecting page data
   Generating static pages (0/197)  [    ]TypeError: Cannot read properties of undefined (reading 'featuredImage')
    at Post (/Users/neha/open-source/blog-website/.next/server/pages/technology/[slug].js:1:858)
    at Wc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
    at Zc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
    at Z (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at Zc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:481)
    at Z (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at $c (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:78:98)
    at bd (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:77:404)
    at Z (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:217)
    at Zc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:71:479)

Error occurred prerendering page "/technology/[slug]". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of undefined (reading 'featuredImage')
    at Post (/Users/neha/open-source/blog-website/.next/server/pages/technology/[slug].js:1:858)
    at Wc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
    at Zc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
    at Z (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at Zc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:481)
    at Z (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at $c (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:78:98)
    at bd (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:77:404)
    at Z (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:217)
    at Zc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:71:479)
TypeError: Cannot read properties of undefined (reading 'featuredImage')
    at Post (/Users/neha/open-source/blog-website/.next/server/pages/community/[slug].js:1:858)
    at Wc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
    at Zc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
    at Z (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at Zc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:481)
    at Z (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at $c (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:78:98)
    at bd (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:77:404)
    at Z (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:217)
    at Zc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:71:479)

Error occurred prerendering page "/community/[slug]". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of undefined (reading 'featuredImage')
    at Post (/Users/neha/open-source/blog-website/.next/server/pages/community/[slug].js:1:858)
    at Wc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:68:44)
    at Zc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:253)
    at Z (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at Zc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:70:481)
    at Z (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:89)
    at $c (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:78:98)
    at bd (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:77:404)
    at Z (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:76:217)
    at Zc (/Users/neha/open-source/blog-website/node_modules/react-dom/cjs/react-dom-server.browser.production.min.js:71:479)
   Generating static pages (13/197) [  ==]
 ✓ Generating static pages (197/197)

> Export encountered errors on following paths:
	/community/[slug]
	/technology/[slug]
```

**Comment by Aditya-eddy** [comment]
Hey @nehagup I think the build errors would be resolved if we close this [issue](https://github.com/keploy/blog-website/pull/7). Then accordingly I would make changes in the code.

**Comment by nehagup** [comment]
> Hey @nehagup I think the build errors would be resolved if we close this [issue](https://github.com/keploy/blog-website/pull/7) first. Then accordingly I would make changes in the code.

Please check if it gets fixed after resolving conflicts. 💯

**Comment by Aditya-eddy** [comment]
Hey @nehagup can you check if it's working correctly now. Thanks!!.

**Review by nehagup — COMMENTED** [review]
`npm run dev` is throwing compiling err 

```

 ✓ Ready in 1742ms
 ○ Compiling /404 ...
 ✓ Compiled /404 in 1301ms (262 modules)
 ✓ Compiled / in 295ms (322 modules)
 ✓ Compiled /community in 194ms (380 modules)
 ✓ Compiled /technology in 92ms (384 modules)
 ⚠ The "images.domains" configuration is deprecated. Please use "images.remotePatterns" configuration instead.
 ⨯ ./pages/technology/[slug].tsx
Error:
  × the name `postBody` is defined multiple times
    ╭─[/Users/neha/open-source/blog-website/pages/technology/[slug].tsx:15:1]
 15 │ import { CMS_NAME } from "../../lib/constants";
 16 │ import PrismLoader from "../../components/prism-loader";
 17 │
 18 │ const postBody = ({ content, post }) => {
    ·       ────┬───
    ·           ╰── previous definition of `postBody` here
 19 │   // Define the regular expression pattern to match the entire URL structure
 20 │   const urlPattern = /https:\/\/keploy\.io\/wp\/author\/[^\/]+\//g;
 21 │
 22 │   // Replace the URL in the content with the desired one using the regular expression
 23 │   const replacedContent = content.replace(
 24 │     urlPattern,
 25 │     `/blog/authors/${post.ppmaAuthorName}/`
 26 │   );
 27 │
 28 │   return replacedContent;
 29 │ };
 30 │
 31 │ const postBody = ({ content, post }) => {
    ·       ────┬───
    ·           ╰── `postBody` redefined here
 32 │   // Define the regular expression pattern to match the entire URL structure
 33 │   const urlPattern = /https:\/\/keploy\.io\/wp\/author\/[^\/]+\//g;
    ╰────
 ○ Compiling /technology/[slug] ...
 ⨯ ./pages/technology/[slug].tsx
Error:
  × the name `postBody` is defined multiple times
    ╭─[/Users/neha/open-source/blog-website/pages/technology/[slug].tsx:15:1]
 15 │ import { CMS_NAME } from "../../lib/constants";
 16 │ import PrismLoader from "../../components/prism-loader";
 17 │
 18 │ const postBody = ({ content, post }) => {
    ·       ────┬───
    ·           ╰── previous definition of `postBody` here
 19 │   // Define the regular expression pattern to match the entire URL structure
 20 │   const urlPattern = /https:\/\/keploy\.io\/wp\/author\/[^\/]+\//g;
 21 │
 22 │   // Replace the URL in the content with the desired one using the regular expression
 23 │   const replacedContent = content.replace(
 24 │     urlPattern,
 25 │     `/blog/authors/${post.ppmaAuthorName}/`
 26 │   );
 27 │
 28 │   return replacedContent;
 29 │ };
 30 │
 31 │ const postBody = ({ content, post }) => {
    ·       ────┬───
    ·           ╰── `postBody` redefined here
 32 │   // Define the regular expression pattern to match the entire URL structure
 33 │   const urlPattern = /https:\/\/keploy\.io\/wp\/author\/[^\/]+\//g;
    ╰────
```

**Comment by Aditya-eddy** [comment]
Hey @nehagup I am bit a new to github and contributing to open source. But I am learning with time with the help of Keploy. If there's more error please give me feedback. Again Thanks.!!

**Pull request merged**